src/api/client.rs

```diff
@@ -139,6 +143,19 @@ impl Client {
                 match &self.signing_key {
                     Some(signing_key) => {
                         self.bearer_token = Some(claims.encode_to(signing_key).await?);
+
+                        #[cfg(not(test))]
```

Comment:
Why are you saving a reqwest client to disk

Reply:
Yeah you should not be saving this information everytime u generate a bearer token, would appreciate context on the motivation behind this if this is necessary

Reply:
its not actually saving the client to disk, internally it's saving the api key as a pem and the user-id to the global config. 
this only ever runs when the signing key needs to be updated, which is also the context in which we would want to re-serialize it anyway. 
the benefit of embedding it in this function is not needing to serialize the client to global config state *every time* we do an API call "just in case" this key got updated. instead, this serialization only occurs when a change actually happens during a `Client` call.

Reply:
all the information needed to generate this bearer token should be on disk already.

this does not run only when the signing key is updated -- it runs on every request

we should never have to sererialize or deserialize this client, the parameters to build it should be configurable

everything you need to generate this token (uuid, ecdsa key, remote config) should be things you can configure at runtime / pull from disk

i dont think i can put a check on this, all this function should do is return a string, and not generate side effects